Source/Relativity.Test.Helpers/Relativity.Test.Helpers/ArtifactHelpers/FoldersHelper.cs

```diff
@@ -32,40 +32,54 @@ public FoldersHelper(IHttpRequestHelper httpRequestHelper)
 
 		public static int GetRootFolderArtifactID(int workspaceID, IServicesMgr svgMgr, string userName, string password)
 		{
-			using (IRSAPIClient client = svgMgr.GetProxy<IRSAPIClient>(userName, password))
+			try
 			{
-				client.APIOptions.WorkspaceID = workspaceID;
-				Query<Folder> query = new Query<Folder>();
-				query.Condition = new TextCondition(FolderFieldNames.Name, TextConditionEnum.EqualTo, WorkspaceHelpers.WorkspaceHelpers.GetWorkspaceName(client, workspaceID));
-				query.Fields = FieldValue.NoFields;
-				var ResultSet = client.Repositories.Folder.Query(query);
-
-				if (!ResultSet.Success)
-				{
-					throw new System.Exception("Folder was not found");
-				}
-				else if (ResultSet.TotalCount == 0)
+				using (IRSAPIClient client = svgMgr.GetProxy<IRSAPIClient>(userName, password))
 				{
-					throw new System.Exception("folder count was 0, so the folder was not found");
+					client.APIOptions.WorkspaceID = workspaceID;
+					Query<Folder> query = new Query<Folder>();
+					query.Condition = new TextCondition(FolderFieldNames.Name, TextConditionEnum.EqualTo, WorkspaceHelpers.WorkspaceHelpers.GetWorkspaceName(client, workspaceID));
+					query.Fields = FieldValue.NoFields;
+					var ResultSet = client.Repositories.Folder.Query(query);
+
+					if (!ResultSet.Success)
+					{
+						throw new TestHelpersException("Folder was not found");
+					}
+					else if (ResultSet.TotalCount == 0)
+					{
+						throw new TestHelpersException("folder count was 0, so the folder was not found");
+					}
+					return ResultSet.Results.FirstOrDefault().Artifact.ArtifactID;
 				}
-				return ResultSet.Results.FirstOrDefault().Artifact.ArtifactID;
+			}
+			catch (Exception exception)
+			{
+				throw new TestHelpersException("Error Getting Folder Name", exception);
 			}
 		}
 
 		public string GetFolderName(int folderArtifactId, int workspaceId)
 		{
-			const string routeName = "GetFolderName";
-
-			var requestModel = new GetFolderNameRequestModel
+			try
 			{
-				FolderArtifactId = folderArtifactId,
-				WorkspaceId = workspaceId
-			};
+				const string routeName = "GetFolderNameAsync";
+
+				var requestModel = new GetFolderNameRequestModel
+				{
+					FolderArtifactId = folderArtifactId,
+					WorkspaceId = workspaceId
+				};
 
-			var responseString = _httpRequestHelper.SendPostRequest(requestModel, routeName);
-			GetFolderNameResponseModel responseModel = JsonConvert.DeserializeObject<GetFolderNameResponseModel>(responseString);
+				var responseString = _httpRequestHelper.SendPostRequest(requestModel, routeName);
+				GetFolderNameResponseModel responseModel = JsonConvert.DeserializeObject<GetFolderNameResponseModel>(responseString);
 
-			return responseModel.FolderName;
+				return responseModel.FolderName;
+			}
+			catch (Exception exception)
+			{
+				throw new TestHelpersException("Error Getting Folder Name", exception);
```

Comment:
can we put the folder name in the exception message. It will be easy to troubleshoot the stack trace.

Reply:
done